### PR TITLE
Feat/dark mode

### PR DESCRIPTION
**Pull request by Manie-oss**

### CODE REVIEW

File: ui/src/components/TopBar/index.tsx [hunk 1]
```diff
@@ -8,8 +9,41 @@ export const TopBar: React.FC<TopBarProps> = ({setSidebarVisibility}) => {
...
+      localStorage.setItem(websiteTheme, lightModeValue);
+      setCurrentTheme(lightModeValue);
+    }
+  };
+
+  function retrieveTheme() {
+    const theme = localStorage.getItem(websiteTheme) || lightModeValue;
+    setCurrentTheme(theme);
+    document.body.classList.add(theme);
+  }
```
Comment (Aksh-Bansal-dev) [inline ui/src/components/TopBar/index.tsx:34]: Use arrow functions. We should have a consistent code style.

**Review by Aksh-Bansal-dev — CHANGES_REQUESTED** [review]
Make the background a little darker wherever we are using SVG background(Login, news carousel).
Also, the cards in the news carousel remain white when we switch to dark mode(which should not happen ideally).

File: ui/src/pages/Login/styles.module.scss [hunk 2]
```diff
@@ -10,6 +10,20 @@
...
+  width: 100vw;
+  background-color: #e2e2e2;
+  background-image: url('../../assets/images/darkbackground.svg');
+  background-size: cover;
+  background-repeat: no-repeat;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: relative;
+}
```
[the remark this answers is not in the document]
Comment (Manie-oss) [reply ui/src/pages/Login/styles.module.scss:25]: this we will be using once the svg for login screen(dark theme) is available on figma.

File: ui/src/components/TopBar/styles.module.scss [hunk 3]
```diff
@@ -39,6 +39,28 @@
...
 }
 
+.themeIcon {
+  height: 32px;
+  width: 32px;
+  background-color: white;
+  border-radius: 20px;
+  border: 2px solid rgb(104, 103, 103);
+  padding: 3px;
+}
```
[the remark this answers is not in the document]
Comment (Manie-oss) [reply ui/src/components/TopBar/styles.module.scss:49]: may I know the reason please!

[the remark this answers is not in the document]
Comment (Aksh-Bansal-dev) [reply ui/src/components/TopBar/styles.module.scss:49]: When we switch to the dark mode, the background of the button remains white, which looks a little odd imo.

[the remark this answers is not in the document]
Comment (Samy-33) [reply ui/src/components/TopBar/styles.module.scss:49]: IMO, it looks good. As yellow object on a black background is weird.

[the remark this answers is not in the document]
Comment (Samy-33) [reply ui/src/components/TopBar/styles.module.scss:49]: @Aksh-Bansal-dev ping.

[the remark this answers is not in the document]
Comment (Aksh-Bansal-dev) [reply ui/src/components/TopBar/styles.module.scss:49]: I was thinking of something like [this](https://i.ytimg.com/vi/x0hGQ3blksE/maxresdefault.jpg).
But it's fine for now, maybe we can change it later.

File: ui/src/pages/Login/styles.module.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (Aksh-Bansal-dev) [reply ui/src/pages/Login/styles.module.scss:25]: I think, for now, we can do something like [this](https://www.figma.com/file/GjOEDxaZ4sT9mBr8A10IiA/Gymkhana?node-id=422%3A2).

[the remark this answers is not in the document]
Comment (Aksh-Bansal-dev) [reply ui/src/pages/Login/styles.module.scss:25]: Please don't resolve conversations on your own. It just makes reviewing more difficult as the reviewer has to check all resolved conversations and check whether they are actually resolved or not.

File: ui/src/components/Layout/Layout.tsx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import classNames from 'classnames';
-import {useState, useRef, useEffect} from 'react';
+import React from 'react';
+import {useState, useRef, useEffect, useContext} from 'react';
```
Comment (Aksh-Bansal-dev) [inline ui/src/components/Layout/Layout.tsx:3]: Are we using `useContext` hook here?

File: ui/src/components/NewsCarousel/styles.module.scss [hunk 5]
```diff
@@ -11,9 +11,11 @@
 .leftContainer {
   width: 30%;
   padding: 5vh 2vw;
+  // background-color: var(--toggle-overlay);
```
Comment (Aksh-Bansal-dev) [inline ui/src/components/NewsCarousel/styles.module.scss:14]: Please remove unnecessary comments.

File: ui/src/components/ProfileCard/index.tsx [hunk 6]
```diff
@@ -5,6 +5,7 @@ import * as z from 'zod';
 import {Input} from '../FormComponents/Input';
 import {Button} from '../Button';
 import {Textarea} from '../FormComponents/Textarea';
+import {useContext} from 'react';
```
Comment (Aksh-Bansal-dev) [inline ui/src/components/ProfileCard/index.tsx:8]: same

File: ui/src/components/Sidebar/Sidebar.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+import React, {useContext} from 'react';
```
Comment (Aksh-Bansal-dev) [inline ui/src/components/Sidebar/Sidebar.tsx:1]: same

File: ui/src/components/TopBar/styles.module.scss [hunk 8]
```diff
@@ -39,6 +39,14 @@
   margin-left: 1vh;
 }
 
+.themeIcon {
+  height: 32px;
+  width: 32px;
+  border-radius: 20px;
+  // border: 2px solid rgb(104, 103, 103);
```
Comment (Aksh-Bansal-dev) [inline ui/src/components/TopBar/styles.module.scss:46]: remove this as well.

File: ui/src/pages/Home/index.tsx [hunk 9]
```diff
@@ -2,6 +2,7 @@ import {ClubsCarousel} from '../../components/ClubsCarousel';
 import {EventsCarousel} from '../../components/EventsCarousel';
 import {Layout} from '../../components/Layout/Layout';
 import {NewsCarousel} from '../../components/NewsCarousel';
+import './styles.module.scss';
```
Comment (Aksh-Bansal-dev) [inline ui/src/pages/Home/index.tsx:5]: Do we need this now?

File: ui/src/pages/Home/styles.module.scss [hunk 10]
```diff
@@ -0,0 +1 @@
+
```
Comment (Aksh-Bansal-dev) [inline ui/src/pages/Home/styles.module.scss:1]: Please remove empty files.

File: ui/src/components/TopBar/index.tsx [hunk 11]
```diff
@@ -1,15 +1,48 @@
 import styles from './styles.module.scss';
-import {logo} from '../../assets';
+import {logo, moonIcon, sunIcon} from '../../assets';
+import {useEffect, useState} from 'react';
 interface TopBarProps {
   setSidebarVisibility: (isSidebarOpenOnMobile: boolean) => void;
 }
+const LightModeValue = 'light_mode';
+const DarkModeValue = 'dark_mode';
+const WebsiteTheme = 'website_theme';
```
Comment (Aksh-Bansal-dev) [inline ui/src/components/TopBar/index.tsx:9]: All letters should be uppercase. 
```js
const LIGHT_MODE_VALUE = 'light_mode';
```

**Review by Aksh-Bansal-dev — CHANGES_REQUESTED** [review]

File: ui/src/components/Theme/index.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
+import {moonIcon, sunIcon} from '../../assets';
+import {useEffect, useState} from 'react';
+import styles from './styles.module.scss';
+
+const LIGHT_MODE_VALUE = 'light_mode';
+const DARK_MODE_VALUE = 'dark_mode';
+const WEBSITE_THEME = 'website_theme';
+
+export const Theme: any = () => {
```
Comment (Aksh-Bansal-dev) [inline ui/src/components/Theme/index.tsx:9]: Try not to use `any` type unless necessary. You can use `React.FC` here.
Also, `Theme` doesn't sound like a component, let's make it something like `ThemeButton`.

**Review by Aksh-Bansal-dev — CHANGES_REQUESTED** [review]

**Review by Aksh-Bansal-dev — APPROVED** [review]
LGTM

**Comment by Manie-oss** [comment]
> LGTM

LGTMT

**Pull request merged**